ruby/vampires.rb

```diff
@@ -0,0 +1,66 @@
+puts "How many employees will be processed?"
+employee = gets.to_i
+
+vampire = 0
+
+#Loop until employee processing is met.
+until vampire == employee 
+
+	puts "What is your name?"
+	name1 = gets.chomp
+    	if name1 == "Drake Cula"
+    		name2 = false
+    	elsif name1 == "Tu Fang"
+    		name2 = false
+    		else name2 = true
+    	end
+  
+    puts "How old are you? What year were you born?"
+	age = gets.to_i
+	birth_year = gets.to_i
+  		if birth_year + age == 2016
+  			age2 = true
+  			else age2 = false
+  		end
+		
+	puts "Our company cafeteria serves garlic bread. Should we order some for you? (y/n)"
+	garlic = gets.chomp
+    	if garlic == "y"
+    		garlic2 = true
+    		else garlic == "n"
+    			garlic2 = false
+    	end
+	
+	puts "Would you like to enroll in the company’s health insurance? (y/n)"
+  	insurance = gets.chomp
+        if insurance == "y"
+        	insurance2 = true
+        	else insurance == "n"
+       			insurance2 = false
+       	end
+      	
+    puts "List any allergies you might have (Enter 'done' when finsihed):"
+    allergies = gets.chomp
+	    until allergies.include? "sunlight" or allergies.include? "done"
+	      allergies = gets.chomp
+	    end
+	    if allergies.include? "sunlight"
+	    	allergies2 = true
+			else
+		      allergies2 = false
+	    end
+  
+	if !name2
```

Comment:
Lets consider order of operations here. I understand that the instructions may be guiding you to just put in conditions as they are mentioned. However, we could clean this up and make the file much more efficient by analyzing our conditions for optimal calls and runtime.

If the employee is named "Drake Cula" or "Tu Fang" then we know that they are definitely a vampire. If that's the case, is there any need to even run the rest of the code?

Consider nesting your conditions and creating a strong order of operations for optimized logic and runtime.
```
if name == "Drake Cula" || name == "Tu Fang"
   "definitely a vampire"
else
  if age_check
    do something
  elsif garlic_and_insurance_check
    do something
  else
    do something
  end
end
```